packages/@aws-cdk/aws-apigateway/README.md

```diff
@@ -1,2 +1,177 @@
-## The CDK Construct Library for AWS API Gateway
-This module is part of the [AWS Cloud Development Kit](https://github.com/awslabs/aws-cdk) project.
+## CDK Construct Library for Amazon API Gateway
+
+Amazon API Gateway is a fully managed service that makes it easy for developers
+to publish, maintain, monitor, and secure APIs at any scale. Create an API to
+access data, business logic, or functionality from your back-end services, such
+as applications running on Amazon Elastic Compute Cloud (Amazon EC2), code
+running on AWS Lambda, or any web application.
+
+### Defining APIs
+
+APIs are defined as a hierarchy of resources and methods. `addResource` and
+`addMethod` can be used to build this hierarchy. The root resource is
+`api.root`.
+
+For example, the following code defines an API that includes the following HTTP
+endpoints: `ANY /, GET /books`, `POST /books`, `GET /books/{book_id}`, `DELETE /books/{book_id}`.
+
+```ts
+const api = new apigateway.RestApi(this, 'books-api');
+
+api.root.addMethod('ANY');
+
+const books = api.root.addResource('books');
+books.addMethod('GET');
```

Comment:
I can't help but think the `method`s should be an `enum`. What's the reasoning for using `string`s?

Reply:
A string felt a bit more convenient than an enum, but I don't feel strongly about this.

I am also wondering whether it should be `addMethod` or `onMethod`. Thoughts?

Reply:
I think I'll rename to `addMethod`. Feels a bit more accurate. I am considering a higher level API `onHttpGet(route, integration)` which will automatically generate the tree, so that will be the "event driven API" instead of `onMethod`